google_drive/public/js/google_docs_edit.js

```diff
@@ -1,24 +1,117 @@
+
+    var clear_name_button = $('.clear-display-name', element);
+    var save_button = $('.save-button', element);
+    var validation_alert = $('.validation_alert', element);
+    var embed_code_textbox = $('#edit_embed_code', element);
+    var xblock_inputs_wrapper = $('.xblock-inputs', element);
+    var edit_display_name_input = $('#edit_display_name', element);
+    var error_message_div = $('.xblock-editor-error-message', element);
+    var defaultName = edit_display_name_input.attr('data-default-value');
+    var edit_alt_text_input = $('#edit_alt_text', element);
+    var alt_text_item = $('li#alt_text_item', element);
+
+    ToggleClearDefaultName();
+    IsUrlValid();
+
+    $('.clear-display-name', element).bind('click', function() {
+        $(this).addClass('inactive');
+        edit_display_name_input.val(defaultName);
+    });
+
+    edit_display_name_input.bind('keyup', function(){
+        ToggleClearDefaultName();
+    });
+
+    $('#edit_embed_code', element).bind('keyup', function(){
+        IsUrlValid();
+    });
+
+    $('.cancel-button', element).bind('click', function() {
+        runtime.notify('cancel', {});
+    });
+
+    function ToggleClearDefaultName(name, button){
+        if (edit_display_name_input.val() == defaultName){
+            if (!clear_name_button.hasClass('inactive')){
+                clear_name_button.addClass('inactive');
+            }
+        }
+        else {
+            clear_name_button.removeClass('inactive');
+        }
+    }
+
+    function SaveEditing(){
         var data = {
-            'display_name': $('.edit-display-name', element).val(),
-            'embed_code': $('.edit-embed-code', element).val(),
+            'display_name': edit_display_name_input.val(),
+            'embed_code': embed_code_textbox.val(),
+            'alt_text': edit_alt_text_input.val(),
         };
 
-        $('.xblock-editor-error-message', element).html();
-        $('.xblock-editor-error-message', element).css('display', 'none');
+        error_message_div.html();
+        error_message_div.css('display', 'none');
         var handlerUrl = runtime.handlerUrl(element, 'studio_submit');
         $.post(handlerUrl, JSON.stringify(data)).done(function(response) {
             if (response.result === 'success') {
                 window.location.reload(false);
             } else {
-                $('.xblock-editor-error-message', element).html('Error: '+response.message);
-                $('.xblock-editor-error-message', element).css('display', 'block');
+                error_message_div.html('Error: '+response.message);
+                error_message_div.css('display', 'block');
             }
         });
-    });
+    }
 
-    $('.cancel-button', element).bind('click', function() {
-        runtime.notify('cancel', {});
-    });
+    function HideAltTextInput(){
+        edit_alt_text_input.val('');
+        alt_text_item.addClass('covered');
+    }
+
+    function IsUrlValid(){
```

Comment:
Is this function a general utility that might be useful for calendar, presentations, etc.?  Should there be a common/utils package for these types of helpers?